internal/form/handler.go

```diff
@@ -0,0 +1,311 @@
+	Description string `json:"description"`
+	Order       int32  `json:"order" validate:"required"`
+}
+
+type Store interface {
+	Create(ctx context.Context, request Request, userid uuid.UUID) (Form, error)
+	Update(ctx context.Context, id uuid.UUID, request Request, userid uuid.UUID) (Form, error)
+	Delete(ctx context.Context, id uuid.UUID) error
+	GetByID(ctx context.Context, id uuid.UUID) (Form, error)
+	List(ctx context.Context) ([]Form, error)
+}
+
+type QuestionStore interface {
+	Create(ctx context.Context, input question.CreateParams) (question.Question, error)
+	Update(ctx context.Context, input question.UpdateParams) (question.Question, error)
+	Delete(ctx context.Context, formID uuid.UUID, id uuid.UUID) error
+	ListByFormID(ctx context.Context, formID uuid.UUID) ([]question.Question, error)
+}
+
+type Handler struct {
+	logger *zap.Logger
+	tracer trace.Tracer
+
+	validator     *validator.Validate
+	problemWriter *problem.HttpWriter
+
+	Store         Store
+	questionStore QuestionStore
+}
+
+func NewHandler(
+	logger *zap.Logger,
+	validator *validator.Validate,
+	problemWriter *problem.HttpWriter,
+	Store Store,
+	questionStore QuestionStore,
+) *Handler {
+	return &Handler{
+		logger:        logger,
+		tracer:        otel.Tracer("form/handler"),
+		validator:     validator,
+		problemWriter: problemWriter,
+		Store:         Store,
+		questionStore: questionStore,
+	}
+}
+
+func (h *Handler) CreateFormHandler(w http.ResponseWriter, r *http.Request) {
+	traceCtx, span := h.tracer.Start(r.Context(), "CreateFormHandler")
+	defer span.End()
+	logger := logutil.WithContext(traceCtx, h.logger)
+
+	var req Request
+	if err := handlerutil.ParseAndValidateRequestBody(traceCtx, h.validator, r, &req); err != nil {
+		h.problemWriter.WriteError(traceCtx, w, err, logger)
+		return
+	}
+
+	currentUser, ok := user.GetUserFromContext(traceCtx)
+	if !ok {
+		h.problemWriter.WriteError(traceCtx, w, fmt.Errorf("user not authenticated"), logger)
+		return
+	}
+
+	form, err := h.Store.Create(traceCtx, req, currentUser.ID)
+	if err != nil {
+		h.problemWriter.WriteError(traceCtx, w, err, logger)
+		return
+	}
+
+	handlerutil.WriteJSONResponse(w, http.StatusCreated, form)
+}
+
+func (h *Handler) UpdateFormHandler(w http.ResponseWriter, r *http.Request) {
+	traceCtx, span := h.tracer.Start(r.Context(), "UpdateFormHandler")
+	defer span.End()
+	logger := logutil.WithContext(traceCtx, h.logger)
+
+	//get form id
+	formIDStr := r.PathValue("id")
+	formID, err := handlerutil.ParseUUID(formIDStr)
+	if err != nil {
+		h.problemWriter.WriteError(traceCtx, w, err, logger)
+		return
+	}
+
+	var req Request
+	if err := handlerutil.ParseAndValidateRequestBody(traceCtx, h.validator, r, &req); err != nil {
+		h.problemWriter.WriteError(traceCtx, w, err, logger)
+		return
+	}
+
+	//get user
+	currentUser, ok := user.GetUserFromContext(traceCtx)
+	if !ok {
+		h.problemWriter.WriteError(traceCtx, w, fmt.Errorf("user not authenticated"), logger)
```

Comment:
Consider using `internal.ErrNoUserInContext` to centralize error messages in `errors.go` for maintainability
```suggestion
		h.problemWriter.WriteError(traceCtx, w, internal.ErrNoUserInContext, logger)
```